test/workbox-routing/node/Router.mjs

```diff
@@ -0,0 +1,302 @@
+import clearRequire from 'clear-require';
+import sinon from 'sinon';
+import {expect} from 'chai';
+
+import Route from '../../../packages/workbox-routing/lib/Route.mjs';
+import Router from '../../../packages/workbox-routing/lib/Router.mjs';
+import expectError from '../../../infra/utils/expectError';
+import {_private} from '../../../packages/workbox-core/index.mjs';
+
+describe(`[workbox-routing] Router`, function() {
+  const sandbox = sinon.sandbox.create();
+  const {logger} = _private;
+  const MATCH = () => {};
+  const HANDLER = () => {};
+  const METHOD = 'POST';
+  const EXPECTED_RESPONSE_BODY = 'test body';
+
+  beforeEach(async function() {
+    clearRequire.all();
+
+    // Run this in the `beforeEach` hook as well as the afterEach hook due to
+    // a mocha bug where `afterEach` hooks aren't run for skipped tests.
+    // https://github.com/mochajs/mocha/issues/2546
+    sandbox.restore();
+    // Prevent logs in the mocha output.
+    sandbox.stub(logger, 'warn');
+    sandbox.stub(logger, 'log');
+  });
+
+  afterEach(function() {
+    sandbox.restore();
+  });
+
+  describe(`constructor`, function() {
+    it(`should construct without any inputs`, function() {
+      expect(() => {
+        new Router();
+      }).to.not.throw();
+    });
+  });
+
+  describe(`calling addFetchListener()`, function() {
```

Comment:
Can we stick the format of:

- top level describe [<module name>] <Class name>
    - describe <function name>
        - it tests
    - describe <function name>
        - it tests

It something Phil started calling out in reviews and I think it's a nice / good pattern to follow.